src/components/Filter/useFilter.ts

```diff
@@ -0,0 +1,22 @@
+import qs from 'qs';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+export const useFilter = <T>() => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const values = qs.parse(location.search, {
+    ignoreQueryPrefix: true,
+  }) as Partial<T>;
+
+  const onChange = (values: Partial<T>) => {
+    const queryVals = qs.stringify(values, {
+      addQueryPrefix: true,
```

Comment:
How do lists stringify? It might be nicer to have
```tsx
arrayFormat: 'comma',
```

Reply:
10-4!

Reply:
Just pushed up this update